apps/devtools/app/scripts/components/MessageViewer/MessageTimeline/MessageTimeline.tsx

```diff
@@ -0,0 +1,143 @@
+import { Box } from '@chakra-ui/react';
+import { FC, useEffect, useMemo, useRef, useState } from 'react';
+import { Canvas, CanvasRef } from 'react-scribble';
+import { MessageTimelineMeta } from './types';
+import { TimelineDrawer } from './TimelineDrawer';
+import {
+  MessageTimelineProps,
+  RowMeta,
+  MessageSpan as MessageSpanType,
+} from './types';
+import { useTimelineColors } from './colors';
+
+const makeRowMeta = (index: number): RowMeta => ({
+  rowCursor: 0,
+  spans: [],
+  rowIndex: index,
+});
+
+export const MessageTimeline: FC<MessageTimelineProps> = ({
+  messages,
+  startedRecordingAt,
+  activeMessageId,
+  selectQuery,
+}) => {
+  /**
```

Comment:
Let's remove this comment, this was done before the functionality was completed. This is sorta how this works already. We just do not zoom and stuff since that would probably be good for the next iteration.